src/drivers/indexeddb.js

```diff
@@ -126,32 +126,145 @@
         return promise;
     }
 
-    function setItem(key, value, callback) {
-        var self = this;
+    function doTask(store, task) {
+        if (task.action === 'put') {
+            return store.put(task.value, task.key);
+        }
 
-        // Cast the key to a string, as that's all we can set as a key.
-        if (typeof key !== 'string') {
-            window.console.warn(key +
-                                ' used as a key, but it is not a string.');
-            key = String(key);
+        if (task.action === 'delete') {
+            // We use a Grunt task to make this safe for IE and some
+            // versions of Android (including those used by Cordova).
+            // Normally IE won't like `.delete()` and will insist on
+            // using `['delete']()`, but we have a build step that
+            // fixes this for us now.
+            return store.delete(task.key);
+        }
+
+        if (task.action === 'clear') {
+            return store.clear();
         }
+    }
 
+    function _updateDatabase(tasks) {
+        var self = this;
         var promise = new Promise(function(resolve, reject) {
             self.ready().then(function() {
                 var dbInfo = self._dbInfo;
                 var transaction = dbInfo.db.transaction(dbInfo.storeName, 'readwrite');
                 var store = transaction.objectStore(dbInfo.storeName);
 
-                // The reason we don't _save_ null is because IE 10 does
-                // not support saving the `null` type in IndexedDB. How
-                // ironic, given the bug below!
-                // See: https://github.com/mozilla/localForage/issues/161
-                if (value === null) {
-                    value = undefined;
+                for (var i=0; i<tasks.length; i++) {
+                    tasks[i].req = doTask(store, tasks[i]);
                 }
 
-                var req = store.put(value, key);
                 transaction.oncomplete = function() {
+                    resolve();
+                };
+                transaction.onerror = function() {
+                    reject();
+                };
+
+                // The request will be aborted if we've exceeded our storage
+                // space. In this case, we will reject with a specific
+                // "QuotaExceededError".
+                transaction.onabort = function(event) {
+                    var error = event.target.error;
+                    if (error === 'QuotaExceededError') {
+                        reject(error);
+                    }
+                };
+            }).catch(reject);
+        });
+
+        return promise;
+    }
+
+    function _collectBatch() {
+        var batchSize = 0,
```

Comment:
Dunno why the linter didn't catch this, but each `var` should be on its own line.